### DISCUSSION TITLE
baremetal implementaion of ethernet - ping an ip

### DISCUSSION THREAD

**Discussion by Nileena00** [polarfire-soc/polarfire-soc-documentation#464] [Bare metal embedded software]
is there anything to learn and understand the baremetal implementaion of ethernet communication and validate it by pinging the ip?

**Comment by hughbreslin** [comment]
Have a look at the MAC examples [here](https://github.com/polarfire-soc/polarfire-soc-bare-metal-examples/tree/main/driver-examples/mss/mss-ethernet-mac)

**Reply by nileena-dev** [reply]
> Have a look at the MAC examples [here](https://github.com/polarfire-soc/polarfire-soc-bare-metal-examples/tree/main/driver-examples/mss/mss-ethernet-mac)

that is too complicate for me to understand.
please help me to just implement a test for verifying the ethernet working just by pinginf the PC's ip!

**Reply by nileena-dev** [reply]
also what if i only want one ethernet i.e., i am using 8221 instead of vsc8332
